archunit/src/main/java/com/tngtech/archunit/core/importer/ImportOption.java

```diff
@@ -137,4 +149,16 @@ public boolean includes(Location location) {
             return !location.isArchive();
         }
     }
+
+    /**
+     * The {@code maven-compiler-plugin 3.10.0} fixed a bug that made {@code package-info.class}
+     * files hidden from {@code ArchUnit}.
+     * This is a {@link ImportOption} implementation for excluding {@code package-info.class} files.
+     */
+    final class DoNotIncludePackageInfoClasses implements ImportOption {
+        @Override
+        public boolean includes(Location location) {
+            return !location.matches(Pattern.compile(".*package-info\\.class$"));
```

Comment:
We should not call `Pattern.compile(..)` on every `includes(..)` invocation. That is quite inefficient. We should compile the `Pattern` one time into a static constant and then reuse it on every `includes(..)` invocation (which might in the end be called tens or hundreds of thousands of times for big projects).

I was also wondering if we should add a simple `Location.endsWith(..)` public API to save some performance :thinking: But I have realized that we're (ab-)using patterns also for some other simple cases like `MAVEN_TEST_PATTERN`. Don't know if there is an internal optimization that makes this as efficient as `contains`, maybe we should do some measurements at some point (unrelated to this PR :wink:)

Reply:
yeah, I guess I missed this, good catch.
fixed in https://github.com/TNG/ArchUnit/pull/794/commits/90c126f6acba68e0602e316c5d576001ee5f20ce